drivers/include/lis2dh12.h

```diff
@@ -179,28 +207,26 @@ typedef struct {
 } lis2dh12_int_params_t;
 
 /**
- * @brief   Status of INT_SRC register
+ * @brief   LIS2DH12 FIFO modes
  */
-#define LIS2DH12_INT_SRC_XL     (0x01)  /**< X low event has occurred */
-#define LIS2DH12_INT_SRC_XH     (0x02)  /**< X high event has occurred */
-#define LIS2DH12_INT_SRC_YL     (0x04)  /**< Y low event has occurred */
-#define LIS2DH12_INT_SRC_YH     (0x08)  /**< Y high event has occurred */
-#define LIS2DH12_INT_SRC_ZL     (0x10)  /**< Z low event has occurred */
-#define LIS2DH12_INT_SRC_ZH     (0x20)  /**< Z high event has occurred */
-#define LIS2DH12_INT_SRC_IA     (0x40)  /**< 1 if interrupt occurred */
-#endif /* MODULE_LIS2DH12_INT */
+typedef enum {
+    LIS2DH12_FIFO_MODE_BYPASS = 0,      /**< default mode, FIFO is bypassed */
+    LIS2DH12_FIFO_MODE_FIFOMODE,        /**< normal FIFO mode, stops if FIFO is full */
+    LIS2DH12_FIFO_MODE_STREAM,          /**< Stream mode, oldest values get overwritten */
+    LIS2DH12_FIFO_MODE_STREAMtoFIFO,    /**< Stream mode and on interrupt jumps to FIFO mode */
+} lis2dh12_fifo_mode_t;
 
 /**
- * @brief   Status of INT_SRC register
+ * @brief   LIS2DH12 FIFO config values
  */
-#define LIS2DH12_STATUS_XDA     (0x01)  /**< X-axis new data available */
-#define LIS2DH12_STATUS_YDA     (0x02)  /**< Y-axis new data available */
-#define LIS2DH12_STATUS_ZDA     (0x04)  /**< Z-axis new data available */
-#define LIS2DH12_STATUS_ZYXDA   (0x08)  /**< on X-, Y-, Z-axis new data available */
-#define LIS2DH12_STATUS_XOR     (0x10)  /**< X-axis data overrun */
-#define LIS2DH12_STATUS_YOR     (0x20)  /**< Y-axis data overrun */
-#define LIS2DH12_STATUS_ZOR     (0x40)  /**< Y-axis data overrun */
-#define LIS2DH12_STATUS_ZYXOR   (0x80)  /**< on X-, Y-, Z-axis data overrun */
+typedef struct {
+    lis2dh12_fifo_mode_t FIFO_mode; /**< set FIFO mode */
+    uint8_t FIFO_watermark:5;       /**< set the FIFO watermark level */
+    bool FIFO_set_INT2;             /**< sets the FIFO interrupt to INT2, otherwise INT1 */
+    bool FIFO_enable;               /**< enables the FIFO */
```

Comment:
Doesn't `FIFO_mode` being != 0 already imply that the fifo should be enabled?

Reply:
Yes, you are right :smile: